swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -448,7 +448,22 @@ extension Adapter: CallbackHandlerDelegate {
     let resolvers = resetToSystemDNSGettingBindResolvers()
 #endif
 
-    return resolvers
+    var parsedResolvers: [String] = []
+
+    // Normalize addresses to remove any possible scope suffixes
+    resolvers.forEach { stringAddress in
```

Comment:
Is there no iterator-like way of doing this? Like `filter_map` and then `collect`? Would avoid the mutability.